source/github/repo.go

```diff
@@ -16,6 +17,20 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// ConfigBackoff is a wrapper for config that checks if yaml/yml file is valid.
+func (c *client) ConfigBackoff(u *library.User, org, name, ref string) (data []byte, err error) {
```

Comment:
I'm not entirely sure the comment here is accurate so I think we'll have to update this a bit 👍 

At least, I have some concerns that this description could be seen as a bit misleading.

I say this because this `ConfigBackoff` function is currently a wrapper around the `Config` function.

The `Config` function doesn't actually verify the `.yaml`/`.yml` file (a.k.a. pipeline configuration) is valid.

The purpose of these functions is to get the pipeline configuration from the source repo stored on GitHub 🙂 

FWIW:

Validating the pipeline configuration is the responsibility of our compiler and get's called later on in the code base 👍